src/puppetlabs/trapperkeeper/authorization/ring_middleware.clj

```diff
@@ -242,15 +248,25 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
+(schema/defn authorization-check :- AuthResultWithRequest
+  "A function that checks that the request is allowed by the provided rules.
+  Returns a map with the request with auth info added, whether the request is
+  authorized, and a message."
+  [request :- ring/Request
+   rules :- rules/Rules
+   allow-header-cert-info :- schema/Bool]
+  (let [req (add-authinfo request allow-header-cert-info)
+        name (ring/authorized-name req)]
+     (assoc (rules/allowed? rules req name) :req req)))
```

Comment:
If we think the `req` might be useful I think we should make it a part of the existing `AuthorizationResult` rather than create a special schema like this.

That makes sense to me, for the `AuthorizationResult` to include information like the request that was authorized, and maybe later the rule that authorized it.